src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1819,9 +1826,13 @@ object Parsers {
      */
     def defOrDcl(start: Int, mods: Modifiers): Tree = in.token match {
       case VAL =>
-        patDefOrDcl(start, posMods(start, mods), in.getDocComment(start))
+        val mod = atPos(in.skipToken()) { Mod.Val() }
+        val modPos = atPos(start) { mods.withAddedMod(mod) }
```

Comment:
Two comments: 

 - Why the name `modPos`? It reads like "the positon of mod" to me. Maybe use `mods1` instead,  or inline the expression.

- I don't think you need an `atPos(start)` here. The position of the modifiers is set automatically to the union of `mods` and `mod`; so you don't need to set the position manually.

Otherwise LGTM.